source/index.ts

```diff
@@ -153,6 +157,8 @@ is.promise = (value: unknown): value is Promise<unknown> => is.nativePromise(val
 
 is.generatorFunction = isObjectOfType<GeneratorFunction>(TypeName.GeneratorFunction);
 
+is.asyncGeneratorFunction = (value: unknown): value is ((...args: any[]) => Promise<unknown>) => getObjectType(value) === TypeName.AsyncGeneratorFunction;
```

Comment:
```typescript
((...args: any[]) => Promise<unknown>)
```
* is this the correct return type?
* such as there is a `GeneratorFunction` type (like above, L158), isn't there an `AsyncGeneratorFunction`?

Reply:
Unless I'm mistaken, it's an async function, it must return a promise like asyncFunction (old L156, New L162)